### PR TITLE
Count number of braces on a line when checking spec links in test-tidy

### PR DESCRIPTION
**Pull request by asajeffrey**
Allow more than one brace per line when checking spec links in test-tidy.

We had problems caused  by `if unsafe { ... } {`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes have tests.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17193)

### CODE REVIEW

**Comment by asajeffrey** [comment]
IRC chat with @jdm and @nox http://logs.glob.uno/?c=mozilla%23servo&s=6+Jun+2017&e=6+Jun+2017#c690538

**Comment by jdm** [comment]
Can you add a test to python/tidy/servo_tidy_tests/rust_tidy.rs and test_tidy.py?

**Comment by jdm** [comment]
Looks good.

**Comment by asajeffrey** [comment]
This kicked up a couple of issues
* We should skip attributes as well as comments when searching for spec links.
* There were some missing spec links in `components/script/dom/webglrenderingcontext.rs`.

**Comment by asajeffrey** [comment]
IRC conversation with @jdm: http://logs.glob.uno/?c=mozilla%23servo&s=7+Jun+2017&e=7+Jun+2017#c691736

TL;DR: @bors-servo r=jdm
